### PR TITLE
port over the reject command from nushell

### PR DESCRIPTION
**Pull request by stormasm**
completed porting reject...

### CODE REVIEW

**Comment by jntrnr** [comment]
Nice. Can you also add some tests either as examples or to src/tests.rs?

File: src/tests.rs [hunk 1]
```diff
@@ -1187,3 +1187,24 @@ fn comment_skipping_2() -> TestResult {
         "40",
     )
 }
+
+#[test]
+fn command_filter_reject_1() -> TestResult {
+    run_test("[[lang, gems]; [nu, 100]] | reject gems", "{lang: nu}")
```
Comment (jntrnr) [inline src/tests.rs:1193]: Hmm, this feels funny to me. `reject` would remove the "gems" column, but here it's also changing the type from table to record.

Comment (jntrnr) [reply src/tests.rs:1193]: It's possible I'm misreading the test though

Comment (stormasm) [reply src/tests.rs:1193]:
```rust
#[test]
fn command_filter_reject_4() -> TestResult {
    run_test(
        "[[lang, gems, grade]; [nu, 100, a]] | reject gems | to json",
        "[{lang:nu,grade:a}]",
    )
}
```
@jntrnr I tried this one as well inside the tests.rs to no avail...

**Pull request merged**